### PR TITLE
Fix/timeout flow

### PR DESCRIPTION
**Pull request by leodoin**
Waiting for QA to be merged...

### CODE REVIEW

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java [hunk 1]
```diff
@@ -285,26 +284,27 @@ public void onMsgShowOptions(){
...
+        // warning: ninja code ahead!
         if(webView == null) {
             webView = buildWebView();
             webView.loadConsentUIFromUrl(url);
+        //post(Runnable) must be called if loadUrl(String) was already called on the same instance
         } else if(!isNative) {
             webView.post(new Runnable() {
                 @Override
                 public void run() {
                     webView.loadConsentUIFromUrl(url);
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:302]: @leodoin what if we implemented a method like:
```java
void loadConsentUIFromUrlInAnotherThread(webview, url) {
  webView.post {
    webView.loadConsentUIFromUrl(url);
  }
}
```

And we always used `loadConsentUIFromUrlInAnotherThread` instead? It seems to me that, loading the webview in another thread might actually be the way to go.

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java [hunk 2]
```diff
@@ -285,26 +284,27 @@ public void onMsgShowOptions(){
...
     private void loadPm() {
-        loadConsentUI(pmUrl());
         isPmOn = true;
+        loadConsentUI(pmUrl());
     }
 
     private void loadConsentUI(String url){
-        mCountDownTimer = getTimer(defaultMessageTimeOut);
         mCountDownTimer.start();
+        // warning: ninja code ahead!
```
Comment (andresilveirah) [inline cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:293]: I'm just trying to isolate the ninjaness to its own method.

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java [hunk 3]
```diff
@@ -139,6 +137,8 @@ public static ConsentLibBuilder newBuilder(Integer accountId, String property, I
         onConsentUIFinished = b.onConsentUIFinished;
         shouldCleanConsentOnError = b.shouldCleanConsentOnError;
 
+        mCountDownTimer = getTimer(defaultMessageTimeOut);
```
Comment (Suryakantj) [inline cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:140]: @leodoin We should initialize timer after the initialization of defaultMessagTimeout field.

Comment (Suryakantj) [inline cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:140]: Also we should consider closing webview once we are throwing timeout error

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java (hunk 1 above)
Comment (leodoin) [reply cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:302]: @andresilveirah , thanks for the clean code feedback. Much appreciated :)

File: cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java (hunk 3 above)
Comment (leodoin) [reply cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java:140]: Thanks for spotting this @Suryakantj . I'm gonna actually pass ConsentLibBuilder. defaultMessageTimeOut as a param and get rid of this unnecessary state once and for all!

Comment (leodoin) [reply cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java]: @Suryakantj , on any error we sent a callback for closing the current view:
private void onErrorTask(ConsentLibException e){
        this.error = e;
        if(shouldCleanConsentOnError) {
            storeClient.clearIABConsentData();
        }
        cancelCounter();
        closeCurrentMessageView();
        runOnLiveActivityUIThread(() -> GDPRConsentLib.this.onError.run(e));
    }

**Pull request merged**